libsolidity/Types.cpp

```diff
@@ -320,6 +322,14 @@ TypePointer IntegerConstantType::binaryOperatorResult(Token::Value _operator, Ty
 				return TypePointer();
 			value = m_value % other.m_value;
 			break;
+		case Token::Exp:
+			if (other.m_value < 0)
+				return TypePointer();
+			else if (other.m_value > std::numeric_limits<unsigned int>::max())
+				return TypePointer();
+			else
+				value = boost::multiprecision::pow(m_value, other.m_value.convert_to<unsigned int>());
```

Comment:
Hi @chriseth, there's no `pow` function for both `big_int` and `big_int` in `boost::multiprecision`. So I resort to this approach, which is to check whether exponent can be fit into an unsigned int, if it can, `boost` has a function for it right away.